### PR TITLE
[Fortran] Set and get halos in Fortran interface

### PR DESCRIPTION
**Pull request by chovyy**
This pull request might be a little bit more controversial than my previous ones.

So far the Fortran interface doesn't make use of the halo meta data fields. `fs_register_field` has arguments for the halos but within `fs_check_size` (which is used by the write routines) it is always called with `...,0,0,0,0,0,0,0,0`. Further, there is no way to retrieve the stored halos.

I have now added two optional arguments to each write routine: minushalos, plushalos. Example:
```fortran
SUBROUTINE fs_write_int_3d(serializer, savepoint, fieldname, field, minushalos, plushalos)
  TYPE(t_serializer), INTENT(IN)          :: serializer
  TYPE(t_savepoint) , INTENT(IN)          :: savepoint
  CHARACTER(LEN=*)                        :: fieldname
  INTEGER(KIND=C_INT), INTENT(IN), TARGET :: field(:,:,:)
  INTEGER, OPTIONAL  :: minushalos(3), plushalos(3)
```

Why have I done it like this? With [FortranTestGenerator](https://github.com/fortesg/fortrantestgenerator) I want to use the halo fields to store the bounds of Fortran arrays and for that, I just want to pass the results of `LBOUND` and `UBOUND` to the write routines. Example:
```fortran
CALL fs_write_field(serializer, savepoint, "example", example, LBOUND(example), UBOUND(example))
```

To retrieve the halos of an existing field, I have added the function `fs_get_halos` which returns an array of size 8 with the values of iminushalo, iplushalo, jminushalo, jplushalo, kminushalo, kplushalo, lminushalo, lplushalo:
```fortran
FUNCTION fs_get_halos(serializer, fieldname)
  TYPE(t_serializer)    :: serializer
  CHARACTER(LEN=*)      :: fieldname
  INTEGER, DIMENSION(8) :: fs_get_halos
```

I know, this looks a little bit inconsistent and far a away from a perfect design, but I have thought a lot about other options and there was nothing that seemed to be any better.

Within the C interface, I have added the function `serialboxSerializerGetFieldMetainfo2` that returns all the meta data fields that are set by `serialboxSerializerAddField2`. `serialboxSerializerGetFieldMetainfo2` is then used used by `serialboxFortranSerializerGetFieldHalos` in the `FortranWrapper`.

Please tell me, when you think that there are better ways to achieve what I want. Anyway, I am not sure if these halos are just a leftover from the old serialbox and I should use my own meta data fields to store the array bounds. Currently, I'm just doing it like this with [FortranTestGenerator](https://github.com/fortesg/fortrantestgenerator) and my modified version of the old serialbox ([serialbox-ftg](https://github.com/fortesg/serialbox-ftg)).

### CODE REVIEW

File: src/serialbox-fortran/m_serialize.f90 [hunk 1]
```diff
@@ -1041,66 +1120,70 @@ SUBROUTINE fs_write_logical_0d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_logical_0d
 
 
-SUBROUTINE fs_write_logical_1d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_logical_1d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   LOGICAL, INTENT(IN), TARGET :: field(:)
+  INTEGER, OPTIONAL  :: minushalos(1), plushalos(1)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Don't you want to make this `INTENT(IN)`?

File: src/serialbox-fortran/m_serialize.f90 [hunk 2]
```diff
@@ -1041,66 +1120,70 @@ SUBROUTINE fs_write_logical_0d(serializer, savepoint, fieldname, field)
...
 END SUBROUTINE fs_write_logical_1d
 
 
-SUBROUTINE fs_write_logical_2d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_logical_2d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   LOGICAL, INTENT(IN), TARGET :: field(:,:)
+  INTEGER, OPTIONAL  :: minushalos(2), plushalos(2)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 3]
```diff
@@ -1041,66 +1120,70 @@ SUBROUTINE fs_write_logical_0d(serializer, savepoint, fieldname, field)
...
 END SUBROUTINE fs_write_logical_2d
 
 
-SUBROUTINE fs_write_logical_3d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_logical_3d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   LOGICAL, INTENT(IN), TARGET :: field(:,:,:)
+  INTEGER, OPTIONAL  :: minushalos(3), plushalos(3)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 4]
```diff
@@ -1041,66 +1120,70 @@ SUBROUTINE fs_write_logical_0d(serializer, savepoint, fieldname, field)
...
 END SUBROUTINE fs_write_logical_3d
 
 
-SUBROUTINE fs_write_logical_4d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_logical_4d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   LOGICAL, INTENT(IN), TARGET :: field(:,:,:,:)
+  INTEGER, OPTIONAL  :: minushalos(4), plushalos(4)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 5]
```diff
@@ -1130,11 +1213,12 @@ SUBROUTINE fs_write_bool_0d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_bool_0d
 
 
-SUBROUTINE fs_write_bool_1d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_bool_1d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   LOGICAL(KIND=C_BOOL), INTENT(IN), TARGET :: field(:)
+  INTEGER, OPTIONAL  :: minushalos(1), plushalos(1)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 6]
```diff
@@ -1157,11 +1241,12 @@ SUBROUTINE fs_write_bool_1d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_bool_1d
 
 
-SUBROUTINE fs_write_bool_2d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_bool_2d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   LOGICAL(KIND=C_BOOL), INTENT(IN), TARGET :: field(:,:)
+  INTEGER, OPTIONAL  :: minushalos(2), plushalos(2)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 7]
```diff
@@ -1750,11 +1858,12 @@ SUBROUTINE fs_write_double_3d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_double_3d
 
 
-SUBROUTINE fs_write_double_4d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_double_4d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   REAL(KIND=C_DOUBLE), INTENT(IN), TARGET :: field(:,:,:,:)
+  INTEGER, OPTIONAL  :: minushalos(4), plushalos(4)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 8]
```diff
@@ -1721,11 +1827,12 @@ SUBROUTINE fs_write_double_2d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_double_2d
 
 
-SUBROUTINE fs_write_double_3d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_double_3d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   REAL(KIND=C_DOUBLE), INTENT(IN), TARGET :: field(:,:,:)
+  INTEGER, OPTIONAL  :: minushalos(3), plushalos(3)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 9]
```diff
@@ -1694,11 +1798,12 @@ SUBROUTINE fs_write_double_1d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_double_1d
 
 
-SUBROUTINE fs_write_double_2d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_double_2d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   REAL(KIND=C_DOUBLE), INTENT(IN), TARGET :: field(:,:)
+  INTEGER, OPTIONAL  :: minushalos(2), plushalos(2)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 10]
```diff
@@ -1667,11 +1770,12 @@ SUBROUTINE fs_write_double_0d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_double_0d
 
 
-SUBROUTINE fs_write_double_1d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_double_1d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   REAL(KIND=C_DOUBLE), INTENT(IN), TARGET :: field(:)
+  INTEGER, OPTIONAL  :: minushalos(1), plushalos(1)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 11]
```diff
@@ -1614,11 +1716,12 @@ SUBROUTINE fs_write_float_3d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_float_3d
 
 
-SUBROUTINE fs_write_float_4d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_float_4d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   REAL(KIND=C_FLOAT), INTENT(IN), TARGET :: field(:,:,:,:)
+  INTEGER, OPTIONAL  :: minushalos(4), plushalos(4)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 12]
```diff
@@ -1587,11 +1687,12 @@ SUBROUTINE fs_write_float_2d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_float_2d
 
 
-SUBROUTINE fs_write_float_3d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_float_3d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   REAL(KIND=C_FLOAT), INTENT(IN), TARGET :: field(:,:,:)
+  INTEGER, OPTIONAL  :: minushalos(3), plushalos(3)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 13]
```diff
@@ -1560,11 +1659,12 @@ SUBROUTINE fs_write_float_1d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_float_1d
 
 
-SUBROUTINE fs_write_float_2d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_float_2d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   REAL(KIND=C_FLOAT), INTENT(IN), TARGET :: field(:,:)
+  INTEGER, OPTIONAL  :: minushalos(2), plushalos(2)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 14]
```diff
@@ -1533,11 +1631,12 @@ SUBROUTINE fs_write_float_0d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_float_0d
 
 
-SUBROUTINE fs_write_float_1d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_float_1d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   REAL(KIND=C_FLOAT), INTENT(IN), TARGET :: field(:)
+  INTEGER, OPTIONAL  :: minushalos(1), plushalos(1)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 15]
```diff
@@ -1480,11 +1577,12 @@ SUBROUTINE fs_write_long_3d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_long_3d
 
 
-SUBROUTINE fs_write_long_4d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_long_4d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   INTEGER(KIND=C_LONG), INTENT(IN), TARGET :: field(:,:,:,:)
+  INTEGER, OPTIONAL  :: minushalos(4), plushalos(4)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 16]
```diff
@@ -1453,11 +1548,12 @@ SUBROUTINE fs_write_long_2d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_long_2d
 
 
-SUBROUTINE fs_write_long_3d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_long_3d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   INTEGER(KIND=C_LONG), INTENT(IN), TARGET :: field(:,:,:)
+  INTEGER, OPTIONAL  :: minushalos(3), plushalos(3)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 17]
```diff
@@ -1426,11 +1520,12 @@ SUBROUTINE fs_write_long_1d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_long_1d
 
 
-SUBROUTINE fs_write_long_2d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_long_2d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   INTEGER(KIND=C_LONG), INTENT(IN), TARGET :: field(:,:)
+  INTEGER, OPTIONAL  :: minushalos(2), plushalos(2)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 18]
```diff
@@ -1399,11 +1492,12 @@ SUBROUTINE fs_write_long_0d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_long_0d
 
 
-SUBROUTINE fs_write_long_1d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_long_1d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   INTEGER(KIND=C_LONG), INTENT(IN), TARGET :: field(:)
+  INTEGER, OPTIONAL  :: minushalos(1), plushalos(1)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 19]
```diff
@@ -1345,11 +1437,12 @@ SUBROUTINE fs_write_int_3d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_int_3d
 
 
-SUBROUTINE fs_write_int_4d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_int_4d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   INTEGER(KIND=C_INT), INTENT(IN), TARGET :: field(:,:,:,:)
+  INTEGER, OPTIONAL  :: minushalos(4), plushalos(4)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 20]
```diff
@@ -1318,11 +1408,12 @@ SUBROUTINE fs_write_int_2d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_int_2d
 
 
-SUBROUTINE fs_write_int_3d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_int_3d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   INTEGER(KIND=C_INT), INTENT(IN), TARGET :: field(:,:,:)
+  INTEGER, OPTIONAL  :: minushalos(3), plushalos(3)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 21]
```diff
@@ -1291,11 +1380,12 @@ SUBROUTINE fs_write_int_1d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_int_1d
 
 
-SUBROUTINE fs_write_int_2d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_int_2d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   INTEGER(KIND=C_INT), INTENT(IN), TARGET :: field(:,:)
+  INTEGER, OPTIONAL  :: minushalos(2), plushalos(2)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 22]
```diff
@@ -1264,11 +1352,12 @@ SUBROUTINE fs_write_int_0d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_int_0d
 
 
-SUBROUTINE fs_write_int_1d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_int_1d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   INTEGER(KIND=C_INT), INTENT(IN), TARGET :: field(:)
+  INTEGER, OPTIONAL  :: minushalos(1), plushalos(1)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 23]
```diff
@@ -1211,11 +1298,12 @@ SUBROUTINE fs_write_bool_3d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_bool_3d
 
 
-SUBROUTINE fs_write_bool_4d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_bool_4d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   LOGICAL(KIND=C_BOOL), INTENT(IN), TARGET :: field(:,:,:,:)
+  INTEGER, OPTIONAL  :: minushalos(4), plushalos(4)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

File: src/serialbox-fortran/m_serialize.f90 [hunk 24]
```diff
@@ -1184,11 +1269,12 @@ SUBROUTINE fs_write_bool_2d(serializer, savepoint, fieldname, field)
 END SUBROUTINE fs_write_bool_2d
 
 
-SUBROUTINE fs_write_bool_3d(serializer, savepoint, fieldname, field)
+SUBROUTINE fs_write_bool_3d(serializer, savepoint, fieldname, field, minushalos, plushalos)
   TYPE(t_serializer), INTENT(IN)          :: serializer
   TYPE(t_savepoint) , INTENT(IN)          :: savepoint
   CHARACTER(LEN=*)                        :: fieldname
   LOGICAL(KIND=C_BOOL), INTENT(IN), TARGET :: field(:,:,:)
+  INTEGER, OPTIONAL  :: minushalos(3), plushalos(3)
```
Comment (clementval) [inline src/serialbox-fortran/m_serialize.f90]: Same as previous comment

**Comment by chovyy** [comment]
Thanks for the comments @clementval. You are right, I forgot the INTENT(in), but at least I was very consistent. ;-)
I'm gonna change that.

**Comment by havogt** [comment]
launch jenkins

**Pull request merged**